simulator/simple_planning_simulator/test/test_simple_planning_simulator.cpp

```diff
@@ -205,183 +140,142 @@ VehicleKinematicState comToBaselink(const VehicleKinematicState & com)
-
-  for (const auto & vehicle_model_type : vehicle_model_type_vec) {
-    std::cout << "\n\n vehicle model = " << vehicle_model_type << std::endl << std::endl;
-    rclcpp::NodeOptions node_options;
-    node_options.append_parameter_override("initialize_source", "INITIAL_POSE_TOPIC");
-    node_options.append_parameter_override("cg_to_rear_m", COM_TO_BASELINK);
-    node_options.append_parameter_override("vehicle_model_type", vehicle_model_type);
-    const auto sim_node = std::make_shared<SimplePlanningSimulator>(node_options);
-
-    const auto pub_sub_node = std::make_shared<PubSubNode>();
-
-    const float32_t target_vel = 5.0f;
-    const float32_t target_acc = 5.0f;
-    const float32_t target_steer = 0.2f;
-
-    auto _resetInitialpose = [&]() {resetInitialpose(sim_node, pub_sub_node);};
-    auto _sendFwdGear = [&]() {sendGear(GearCommand::DRIVE, sim_node, pub_sub_node);};
-    auto _sendBwdGear = [&]() {sendGear(GearCommand::REVERSE, sim_node, pub_sub_node);};
-    auto _sendCommand = [&](const auto & _cmd) {
-        sendCommand(_cmd, sim_node, pub_sub_node);
-      };
+  node_options.append_parameter_override("wheel_radius", 0.5);
+  node_options.append_parameter_override("wheel_width", 0.2);
+  node_options.append_parameter_override("wheel_base", 3.0);
+  node_options.append_parameter_override("wheel_tread", 2.0);
+  node_options.append_parameter_override("front_overhang", 1.0);
+  node_options.append_parameter_override("rear_overhang", 1.0);
+  node_options.append_parameter_override("left_overhang", 0.5);
+  node_options.append_parameter_override("right_overhang", 0.5);
+  node_options.append_parameter_override("vehicle_height", 1.5);
+}
 
-    // check pub-sub connections
-    {
-      size_t expected = 1;
-      EXPECT_EQ(pub_sub_node->pub_control_command_->get_subscription_count(), expected);
-      EXPECT_EQ(pub_sub_node->pub_ackermann_command_->get_subscription_count(), expected);
-      EXPECT_EQ(pub_sub_node->pub_state_cmd_->get_subscription_count(), expected);
-      EXPECT_EQ(pub_sub_node->pub_initialpose_->get_subscription_count(), expected);
-      EXPECT_EQ(pub_sub_node->kinematic_state_sub_->get_publisher_count(), expected);
-    }
+void testVehicleMotion(const std::string & vehicle_model_type)
+{
+  rclcpp::init(0, nullptr);
 
-    // check initial pose
-    _resetInitialpose();
-    const auto init_state = *(pub_sub_node->current_state_);
+  std::cout << "\n\n vehicle model = " << vehicle_model_type << std::endl << std::endl;
+  rclcpp::NodeOptions node_options;
+  node_options.append_parameter_override("initialize_source", "INITIAL_POSE_TOPIC");
+  node_options.append_parameter_override("vehicle_model_type", vehicle_model_type);
+  node_options.append_parameter_override("initial_engage_state", true);
+  node_options.append_parameter_override("add_measurement_noise", false);
+  declareVehicleInfoParams(node_options);
+  const auto sim_node = std::make_shared<SimplePlanningSimulator>(node_options);
```

Comment:
FYI;

When using `launch_testing`, you can pass parameters to Node as follows:
```python
@pytest.mark.launch_test
def generate_test_description():
    return launch.LaunchDescription([
        launch_ros.actions.Node(
            package="simple_planning_simulator",
            executable="simple_planning_simulator_exe",
            name="simple_planning_simulator_node",
            parameters=[{
                "initialize_source": "INITIAL_POSE_TOPIC",
                "vehicle_model_type": vehicle_model_type,
                # etc...
            }],
        ),
        launch_testing.actions.ReadyToTest()
    ])
```